chapters/planning-data-work.tex

```diff
@@ -1,147 +1,459 @@
+Also consider the cost of licenses, the time to learn new tools,
+and the stability of the tools.
+There are few strictly right or wrong choices for software,
+but what is important is that you have a plan in advance
+and understand how your tools will interact with your work.
+
+Ultimately, the goal is to ensure that you will be able to hold
+your code environment constant over the lifecycle of a single project.
+While this means you will inevitably have different projects
+with different code environments, each one will be better than the last,
+and you will avoid the extremely costly process of migrating a project
+into a new code enviroment while it is still ongoing.
+This can be set up down to the software level:
+you should ensure that even specific versions of software
+and the individual packages you use
+are referenced or maintained so that they can be reproduced going forward
+even if their most recent releases contain changes that would break your code.
+  \index{software versions}
+(For example, our command \texttt{ieboilstart} in the \texttt{ietoolkit} package
+provides functionality to support Stata version stability.\sidenote{
+  \url{https://dimewiki.worldbank.org/wiki/ieboilstart}})
+
+Next, think about how and where you write and execute code.
+This book focuses mainly on primary survey data,
+so we are going to broadly assume that you are using ``small'' datasets
+in one of the two most popular desktop-based packages: R or Stata.
+(If you are using another language, like Python,
+or working with big data projects on a server installation,
+many of the same principles apply but the specifics will be different.)
+The most visible part of working with code is a code editor,
+since most of your time will be spent writing and re-writing your code.
+This does not need to be the same program as the code runs in,
+and the various members of your team do not need to use the same editor.
+Using an external editor can be preferable since your editor will not crash if your code does,
+and may offer additional features aimed at writing code well.
+If you are working in R, \textbf{RStudio} is the typical choice.\sidenote{
+  \url{https://www.rstudio.com}}
+For Stata, the built-in do-file editor is the most widely adopted code editor,
+but \textbf{Atom}\sidenote{\url{https://atom.io}} and
+\textbf{Sublime}\sidenote{\url{https://www.sublimetext.com/}}
+can also be configured to run Stata code externally,
+while offering great code accessibility and quality features.
+(We recommend setting up and becoming comfortable with one of these.)
+For example, these editors can access an entire directory -- rather than a single file --
+which gives you access to directory views and file management actions,
+such as folder management, Git integration,
+and simultaneous work with other types of files, without leaving the editor.
+
+In our field of development economics,
+Stata is by far the most commonly used programming language,
```

Comment:
I do not know about `by far`. That is a statement that is likely to be outdated very soon.

Reply:
https://aeadataeditor.github.io/presentation-20191211/#9
🤷
aab065bb7d0259b21614ce8565f411c36fbebcbc

Reply:
ok, you might have the stats on your side, but I like the word _currently_ much better. At least it does not seem as much as if we are promoting it.